src/device_id.c

```diff
@@ -40,3 +55,55 @@ astarte_result_t astarte_device_id_generate_deterministic(
 
     return uuid_to_base64url(uuid, out, ASTARTE_DEVICE_ID_LEN + 1);
 }
+
+astarte_result_t astarte_device_id_generate_unique(
```

Comment:
Is it a duplicate of this ?  https://github.com/astarte-platform/astarte-device-sdk-esp32/blob/05a36a607fda404a5f86248a6b76dbfad87e73a5/src/astarte_hwid.c#L27

Do you move it to this file?

Reply:
Yes, `astarte_hwid.c` has already been removed from the source tree in `api-restructuring`.
The function is very similar to `astarte_hwid_get_id` but more in line with the new APIs.
And with a better name, as its purpose is to generate a new device ID, not a generic hardware ID.